aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/util/CognitoAuthExceptionConverter.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.auth.cognito.util;
+
+import com.amplifyframework.auth.AuthException;
+
+import com.amazonaws.services.cognitoidentityprovider.model.AliasExistsException;
+import com.amazonaws.services.cognitoidentityprovider.model.CodeDeliveryFailureException;
+import com.amazonaws.services.cognitoidentityprovider.model.CodeMismatchException;
+import com.amazonaws.services.cognitoidentityprovider.model.ExpiredCodeException;
+import com.amazonaws.services.cognitoidentityprovider.model.InvalidParameterException;
+import com.amazonaws.services.cognitoidentityprovider.model.InvalidPasswordException;
+import com.amazonaws.services.cognitoidentityprovider.model.LimitExceededException;
+import com.amazonaws.services.cognitoidentityprovider.model.PasswordResetRequiredException;
+import com.amazonaws.services.cognitoidentityprovider.model.ResourceNotFoundException;
+import com.amazonaws.services.cognitoidentityprovider.model.TooManyFailedAttemptsException;
+import com.amazonaws.services.cognitoidentityprovider.model.UserNotConfirmedException;
+import com.amazonaws.services.cognitoidentityprovider.model.UserNotFoundException;
+import com.amazonaws.services.cognitoidentityprovider.model.UsernameExistsException;
+
+/**
+ * Convert AWS Cognito Exceptions to AuthExceptions.
+ */
+public final class CognitoAuthExceptionConverter {
+
+    /**
+     * Dis-allows instantiation of this class.
+     */
+    private CognitoAuthExceptionConverter() {}
+
+    /**
+     * Lookup method to convert AWS Cognito Exception to AuthException.
+     * @param fallbackMessage Fallback message to inform failure
+     * @param error Exception thrown by AWSMobileClient
+     * @return AuthException
+     */
+    public static AuthException lookup(String fallbackMessage, Exception error) {
+        if (error instanceof UserNotFoundException) {
+            return new AuthException.UserNotFoundException(error);
+        }
+
+        if (error instanceof UserNotConfirmedException) {
+            return new AuthException.UserNotConfirmedException(error);
+        }
+
+        if (error instanceof UsernameExistsException) {
+            return new AuthException.UsernameExistsException(error);
+        }
+
+        if (error instanceof AliasExistsException) {
+            return new AuthException.AliasExistsException(error);
```

Comment:
I missed if you replied to this on the earlier PR but why do we need Alias Exists in addition to UsernameExists?